Breadcrumbs.py

```diff
@@ -32,80 +38,169 @@ def get_row_indentation(points, view, tab_size, limit=1e20):
+      previous_length = current_length
+      number_of_not_shorter = len(breadcrumbs) - number_of_shorter
+      if number_of_characters_left < current_length * number_of_not_shorter:
+        length_of_short_trim = number_of_characters_left // number_of_not_shorter
+        length_of_long_trim = length_of_short_trim + 1
+        number_of_long_trimmed = number_of_characters_left % number_of_not_shorter
+        number_of_short_trimmed = number_of_not_shorter - number_of_long_trimmed
+        for (index_of_breadcrumb, breadcrum_length) in enumerate(lengths):
+          if current_length <= breadcrum_length:
+            if 0 < number_of_short_trimmed:
+              length_of_trim = length_of_short_trim
+              number_of_short_trimmed -= 1
+            else:
+              length_of_trim = length_of_long_trim
+            breadcrumbs[index_of_breadcrumb] = breadcrumbs[index_of_breadcrumb][0:length_of_trim]
+        break
+    number_of_characters_left -= current_length
+
+  return ''.join(breadcrumbs+[''])
+
+
+stylesheet = '''
+  <style>
+    div.phantom-arrow {
+      border-top: 0.4rem solid transparent;
+      border-left: 0.5rem solid color(var(--bluish) blend(var(--background) 30%));
+      width: 0;
+      height: 0;
+    }
+    div.phantom {
+      padding: 0.4rem 0 0.4rem 0.7rem;
+      margin: 0 0 0.2rem;
+      border-radius: 0 0.2rem 0.2rem 0.2rem;
+      background-color: color(var(--bluish) blend(var(--background) 30%));
+    }
+    div.phantom span.message {
+      padding-right: 0.7rem;
+    }
+    div.phantom a {
+      text-decoration: inherit;
+    }
+    div.phantom a.close {
+      padding: 0.35rem 0.7rem 0.45rem 0.8rem;
+      position: relative;
+      bottom: 0.05rem;
+      border-radius: 0 0.2rem 0.2rem 0;
+      font-weight: bold;
+    }
+    html.dark div.phantom a.close {
+      background-color: #00000018;
+    }
+    html.light div.phantom a.close {
+      background-color: #ffffff18;
+    }
+  </style>
+'''
+
+
+template = '''
+  <body>
+    {stylesheet}
+    <div class="phantom-arrow"></div>
+    <div class="phantom">
+      <span class="message">
+        <strong>Breadcrumbs:</strong> {breadcrumbs}
+        <a class="close" href="close">''' + chr(0x00D7) + '''</a>
+      </span>
+    </div>
+  </body>
+'''
+
+
+class BreadcrumbsCommand(sublime_plugin.EventListener):
+
+  def on_selection_modified_async(self, view):
+    if settings.get('breadcrumbs_statusbar', True):
+      view.set_status('breadcrumbs', make_breadcrumbs(view))
+    else:
+      view.erase_status('breadcrumbs')
+
+
+class BreadcrumbsPhantomCommand(sublime_plugin.TextCommand):
+
+  def __init__(self, view):
+    self.view = view
+    self.phantom_set = sublime.PhantomSet(view, 'breadcrumbs')
+
+  def on_phantom_close(self, href):
+    self.view.erase_phantoms('breadcrumbs')
+
+  def run(self, edit):
+    phantoms = []
+    self.view.erase_phantoms('breadcrumbs')
+
+    for region in self.view.sel():
```

Comment:
> May I suggest the following idea?

Yes you may ;) Good suggestions. I can also have a peek at how git-blame deals with this. Shouldn't be too hard.